### PR TITLE
Makes final warn that is not supported instead of throwing

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/pinojs/pino-pretty/issues/37

### CODE REVIEW

**Comment by davidmarkclements** [comment]
I've read up on the issue - not sure I agree that warning is better than throwing. 

Throwing is the right thing because you shouldn't use pino.final at all with streams 

I'd rather pino-pretty catches the error and then prints a warning itself

**Comment by davidmarkclements** [comment]
yeah I think it caters to the two targeted cases - 

1. output a warning when using pretty print with pino.final
2. throw when using any other stream with `pino.final`

**Comment by mcollina** [comment]
PTAL

File: test/final.test.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ test('throws if the supplied handler is not a function', async ({ throws }) => {
 test('throws if not supplied logger with pino.destination or pino.extreme instance', async ({ throws, doesNotThrow }) => {
   throws(() => {
     pino.final(pino(fs.createWriteStream('/dev/null')), () => {})
-  }, Error('only compatible with loggers with pino.destination and pino.extreme'))
+  }, Error('inal requires a stream that has a flushSync method, such as pino.destination and pino.extreme'))
```
Comment (davidmarkclements) [inline test/final.test.js]: typo: `inal` -> `final`

**Comment by davidmarkclements** [comment]
other than the typo - LGTM

**Pull request merged**